components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -18,64 +18,63 @@ export default function ShoppingListEditor({ items, onDelete, onAdd }) {
   }, []);
 
   //open drop down when typing into input field
-  useEffect(() => {
-    triggerDropDown();
-  }, [itemInput]);
-
-  //reset dropDownItems state, to remove dropDown buttons from screen,...
-  //...when shopping list (items) changed
-  useEffect(() => {
-    setItemInput("");
-    setDropDownItems([]);
-  }, [items]);
-
-  //...when loosing focus of input field
-  //-> setDropDownReset state needed to allow clicking drop down buttons,
-  //which also removes focus on input field
-  useEffect(() => {
-    if (dropDownReset) {
-      setDropDownItems([]);
-      setDropDownReset(false);
-    }
-  }, [dropDownReset]);
+  function handleItemInput(event) {
+    const inputString = event.target.value;
+    setItemInput(inputString);
+    triggerDropDown(inputString);
+  }
 
   //evoke rendering drop down buttons for items that match input
-  function triggerDropDown() {
-    const matchedDropDownItems = matchInput(itemInput);
+  function triggerDropDown(inputString) {
+    const matchedDropDownItems = matchInput(inputString);
     setDropDownItems(matchedDropDownItems);
   }
 
   //match input with all available items from DB
   function matchInput(value) {
-    const editedValue = value.trim().toLowerCase(); //trim() -> remove white spaces at beginning and end of the name
+    const editedValue = value.trim().toLowerCase();
     //clear drop down when clearing input field
     if (editedValue === "") {
       return [];
     }
-    const matchedItems = allItems.filter((item) =>
+
+    //filter out items from allItems, that are already on the shopping list
+    const usedItemIds = items.map((usedItem) => usedItem.item_id);
+    const availableItems = allItems.filter(
+      (DBitem) => !usedItemIds.includes(DBitem.id)
+    );
+
+    const matchedItems = availableItems.filter((item) =>
       item.name.toLowerCase().startsWith(editedValue)
     );
-    //filter out matched items, that are already on the shopping list
-    const uniqueMatchedItems = matchedItems.filter((matchedItem) => {
-      return (
-        items.find((item) => matchedItem.id === item.item_id) === undefined
-      );
-    });
-    return uniqueMatchedItems;
+    return matchedItems;
   }
 
-  function onSubmit(event) {
-    event.preventDefault();
-    event.target.reset();
-    inputRef.current.focus();
+  function onAddReset() {
+    setItemInput("");
+    setDropDownItems([]);
+    setIsFocusInput(true); //to focus item input after adding item to list
+  }
```

Comment:
As mentioned above, you could write a function handleDropDownAdd which looks like this:
```js
function handleDropDownAdd(element) {
onAdd(element) // the function from the props
 setItemInput("");
 setDropDownItems([]);
 setIsFocusInput(true); 
}



```